src/plugins/intel_cpu/src/emitters/tpp/x64/jit_debug_emitter.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2020-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "jit_tpp_emitter.hpp"
+#include "emitters/plugin/x64/jit_emitter.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+/**
+ * @interface DebugTppEmitter
+ * @brief The purpose of this emitter is to facilitate debugging of TPP emitters. It allows to access attributes of the
+ * owned Tpp emitter at runtime, inspect source ov::Node, in/out memory before and after execution, etc.
+ */
+class DebugTppEmitter : public TppEmitter {
+public:
+    DebugTppEmitter(const ov::snippets::lowered::ExpressionPtr& expr, const std::shared_ptr<TppEmitter>& original)
+            : TppEmitter(*original),
+            m_original(original),
+            m_compiled_kernel(m_original->get_compiled_kernel_ptr()),
+            m_execute_function(m_original->get_execute_function_ptr()),
+            m_source_node(expr->get_node()) {
```

Comment:
I don't usually use it this way, since `friendly name` and node type are the most informative info, but it's a valid point, let's save the whole expression.